dds/DCPS/Recorder.cpp

```diff
@@ -21,6 +21,21 @@ Recorder::~Recorder()
 {
 }
 
+XTypes::DynamicData Recorder::get_dynamic_data(const RawDataSample& sample) {
+
+  Encoding enc(Encoding::KIND_XCDR2, sample.header_.byte_order_ ? ENDIAN_LITTLE : ENDIAN_BIG);
```

Comment:
The `Encoding` object should be read from the encapsulation header to make sure the encoding is correct.